db/log_reader.cc

```diff
@@ -450,17 +461,49 @@ unsigned int Reader::ReadPhysicalRecord(Slice* result, size_t* drop_size) {
 
     buffer_.remove_prefix(header_size + length);
 
-    *result = Slice(header + header_size, length);
-    return type;
+    if (uncompress_ && type != kSetCompressionType) {
+      // Uncompress compressed records
+      uncompressed_record_.clear();
+      size_t uncompressed_size = 0;
+      int remaining = 0;
+      do {
```

Comment:
While writing, a compressed chunk may be spilled into multiple physical records. However, while reading, we uncompress one physical record at a time. Is this asymmetry ok?

Reply:
Yes, the compressed and uncompressed chunks need not be aligned. In case of uncompress, there are partial results that are appended to get the final chunk when the last record is uncompressed.